src/hotspot/share/opto/compile.hpp

```diff
@@ -1058,6 +1058,10 @@ class Compile : public Phase {
           int is_fancy_jump, bool pass_tls,
           bool return_pc, DirectiveSet* directive);
 
+  ~Compile() {
+    delete _print_inlining_stream;
+  };
+
```

Comment:
compile.cpp has print_inlining_stream_free() calls which will leak the stringStream now if called.  I think this function needs to be removed and it should call the reset function to reinitialize the stream.
There should be compiler tests that will fail if print_inlining_stream_free() is called with a null _print_inlining_stream pointer (I think the delete should fail (?) with null)

Reply:
This is removed as part of https://github.com/openjdk/jdk/pull/10396 . The function used to be here but I merged with upstream when that went in. delete on null is OK, just like with free.

Reply:
Ok, I see it removed now.